api/app/Http/Controllers/BaseController.php

```diff
@@ -1,69 +1,66 @@
 <?php namespace App\Http\Controllers;
 
-use App;
+use App\Http\Transformers\CollectionTransformerInterface;
+use App\Http\Transformers\ItemTransformerInterface;
+use App\Repositories\BaseRepository;
+use App\Services\Validator;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Laravel\Lumen\Routing\Controller;
-use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+use Symfony\Component\HttpFoundation\Response;
 use Illuminate\Http\Request;
-use Illuminate\Database\Eloquent\Model;
-use Illuminate\Database\Eloquent\Collection;
 
 abstract class BaseController extends Controller
 {
-    public static $model;
-
     /**
      * Model Repository.
      *
-     * @var App\Repositories\BaseRepository
+     * @var BaseRepository
      */
     protected $repository;
 
     /**
      * Validation service.
      *
-     * @var App\Services\Validation\TestValidator
+     * @var Validator
      */
     protected $validator;
 
-    /**
-     * Transformer to use for responses.
-     *
-     * @var string
-     */
-    protected $transformer = 'App\Http\Transformers\BaseTransformer';
 
     /**
      * Get all entities.
      *
+     * @param CollectionTransformerInterface $transformer
      * @return Response
      */
-    public function getAll()
+    public function getAll(CollectionTransformerInterface $transformer)
     {
-        return $this->collection($this->repository->all());
+        return response($transformer->transformCollection($this->repository->all()));
     }
 
     /**
      * Get one entity.
      *
-     * @param  string  $id
+     * @param ItemTransformerInterface $transformer
+     * @param  string $id
      * @return Response
      */
-    public function getOne($id)
+    public function getOne(ItemTransformerInterface $transformer,$id)
     {
-        return $this->item($this->repository->find($id));
+        return response($transformer->transformItem($this->repository->find($id)));
     }
 
     /**
      * Post a new entity.
      *
      * @param  Request $request
-     * @return mixed
+     * @return Response
      */
     public function postOne(Request $request)
     {
         $this->validator->with($request->all())->validate();
-
-        return response($this->repository->create($request->all()), 201);
+        $model = $this->repository->getModel();
+        $model->fill($request->all());
+        return response((bool)$this->repository->save($model), 201);
```

Comment:
What does casting the return of `$this->repository->save` do? If it is unsuccessful I assume that it throws an exception, so it would make more legibility sense to have the line read
```
$this->repository->save($model);
return response(true, 201);
```